client/src/App.js

```diff
@@ -1,18 +1,25 @@
-import React, { useState } from 'react';
+import React, { useState } from "react";
+import { Route } from "react-router-dom";
 
-import SavedList from './Movies/SavedList';
+import SavedList from "./Movies/SavedList";
+import Movie from "./Movies/Movie";
+import MovieList from "./Movies/MovieList";
 
 const App = () => {
-  const [savedList, setSavedList] = useState( [] );
+  const [savedList, setSavedList] = useState([]);
 
   const addToSavedList = movie => {
-    setSavedList( [...savedList, movie] );
+    !savedList.includes(movie) && setSavedList([...savedList, movie]);
   };
 
   return (
     <div>
       <SavedList list={savedList} />
-      <div>Replace this Div with your Routes</div>
+      <Route exact path="/" component={MovieList} />
+      <Route
+        path="/movies/:id"
+        render={props => <Movie {...props} addToSavedList={addToSavedList} />}
```

Comment:
on a simple level what's some differences between render vs component?

Reply:
with render the type of element that render prop function returns will not change on each render, even though a new anonymous function instance is created on each render.